cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -666,5 +652,15 @@ func main() {
 			logging.Errorf("Failed to notify systemd of readiness: %v", err)
 		}
 	}()
-	proxyClient.Run(connSrc)
+	proxyClient.RunContext(ctx, connSrc)
+	if err, ok := <-shutdown; ok {
+		return err
+	}
+	return nil
+}
+
+func main() {
+	if err := startProxy(); err != nil {
+		os.Exit(1)
```

Comment:
Some of the other `os.Exits` return different error codes return a 2 - this would possible be a breaking change?

Reply:
I'll update the `startProxy` function to return an exit code, to maintain the different error codes.